lib/AST/ASTContext.cpp

```diff
@@ -518,6 +518,11 @@ ASTContext::ASTContext(LangOptions &langOpts, SearchPathOptions &SearchPathOpts,
 }
 
 ASTContext::~ASTContext() {
+  // Emit evaluator dependency graph if requested.
+  auto graphPath = LangOpts.RequestEvaluatorGraphVizPath;
+  if (!graphPath.empty()) {
+    evaluator.emitRequestEvaluatorGraphViz(graphPath);
+  }
```

Comment:
I'm not thrilled with triggering this from the `ASTContext` destructor. I'd prefer that it be triggered from `FrontendTool.cpp` where it was triggered before, and that this option remain a `FrontendOption` (not a `LangOption`). If we need to add another call to `emitRequestEvaluatorGraphViz` from `swift-ide-test` or other tools, that's fine.